src/components/AuthenticatedContributeMeasurementForm.tsx

```diff
@@ -147,7 +149,30 @@ export const AuthenticatedForm: FC<AuthenticatedFormProps> = (props) => {
     undefined
   );
 
+  const checkIfNodeValid = async() => {
```

Comment:
Is it possible to move this inside the useEffect or does that affect how it runs?

Reply:
Yep, done. out of curiosity - What would be a reason for putting a function declaration within the useEffect though ? Wouldn't this cause it to be created again every time there is a re-render ?